src/app/components/registration/registration.component.ts

```diff
@@ -19,15 +20,37 @@ export class RegistrationComponent implements OnInit {
     password: new FormControl(''),
   });
 
-  constructor(private formBuilder: FormBuilder) {}
+  get regPassword() {
+    return this.regForm.get('password');
+  }
+  constructor(private formBuilder: FormBuilder, private http: HttpClient) {}
 
   ngOnInit(): void {
     this.regForm = this.formBuilder.group({
       firstName: ['', [Validators.required]],
       lastName: ['', [Validators.required]],
-      email: ['', [Validators.required]],
-      password: ['', [Validators.required]],
+      email: ['', [Validators.required, Validators.email]],
+      password: [
+        '',
+        [
+          Validators.required,
+          Validators.pattern(RegExp('(?=.*[0-9])(?=.*[A-Z])')),
+        ],
+      ],
+
+      // bandyta '(?=.*?[0-9])(?=.*?[A-Z])$' bet kazkas negeraiiiiii
     });
   }
-  onSubmit() {}
+
+  onSubmit({ data }: { data: any }) {
+    this.http
```

Comment:
http post should be inside service and not in component

Reply:
It is fixed in other branch - "user-authentication", which is not yet merged.